crates/sui-types/src/temporary_store.rs

```diff
@@ -1160,8 +1161,10 @@ impl<'backing> BackingPackageStore for TemporaryStore<'backing> {
             self.store.get_package_object(package_id).map(|obj| {
                 // Track object but leave unchanged
                 if let Some(v) = obj.clone() {
-                    // Can this lock ever block execution?
-                    self.runtime_read_objects.write().insert(*package_id, v);
+                    // TODO: Can this lock ever block execution?
```

Comment:
Yes, this lock can block execution, because a package can be loaded in response to a function call to a package that has not yet been loaded in the epoch.